lib/public/views/Tags/Overview/TagsOverviewModel.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+import { fetchClient, Observable, RemoteData } from '/js/src/index.js';
+import { remoteDataFromResponse } from '../../../utilities/remoteDataFromResponse.js';
+
+/**
+ * Model for tags overview page
+ */
+export class TagsOverviewModel extends Observable {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+
+        /**
+         * @type {RemoteData}
+         * @private
+         */
+        this._tagsList = RemoteData.notAsked();
+
+        /**
+         * Filters to apply on tags list fetching
+         * @type {Object}
+         * @private
+         */
+        this._filters = {};
+    }
+
+    /**
+     * Fetch the tags list to display, applying current filters
+     *
+     * @return {void}
+     */
+    async fetchTagsList() {
+        this._tagsList = RemoteData.loading();
+
+        const params = {
+            ...this._filters.partialText && {
+                'filter[partialText]': this._filters.partialText,
+            },
+        };
+
+        const endpoint = `/api/tags?${new URLSearchParams(params).toString()}`;
+        const response = await fetchClient(endpoint, { method: 'GET' });
+
+        const { remoteData } = await remoteDataFromResponse(response);
+        this._tagsList = remoteData;
+        this.notify();
+    }
+
+    /**
+     * Reset the model to its initial state
+     *
+     * @return {void}
+     */
+    reset() {
+        this._tagsList = RemoteData.notAsked();
```

Comment:
Same call for `this.notify()` I think is needed here as well

Reply:
Ticket created here: https://alice.its.cern.ch/jira/projects/O2B/issues/O2B-619